src/account/default.ts

```diff
@@ -305,11 +305,34 @@ export class Account extends Provider implements AccountInterface {
   }
 
   public async execute(
-    calls: AllowArray<Call>,
-    abis: Abi[] | undefined = undefined,
-    details: UniversalDetails = {}
+    transactions: AllowArray<Call>,
+    transactionsDetail?: UniversalDetails
+  ): Promise<InvokeFunctionResponse>;
+  public async execute(
+    transactions: AllowArray<Call>,
+    abis?: Abi[],
+    transactionsDetail?: UniversalDetails
+  ): Promise<InvokeFunctionResponse>;
+  public async execute(
+    transactions: AllowArray<Call>,
+    arg2: Abi[] | UniversalDetails | undefined = undefined,
+    transactionsDetail: UniversalDetails = {}
   ): Promise<InvokeFunctionResponse> {
-    const transactions = Array.isArray(calls) ? calls : [calls];
+    let details: UniversalDetails;
+    switch (true) {
+      case Array.isArray(arg2):
+        details = transactionsDetail;
+        break;
+      case typeof arg2 === 'object':
+        details = arg2;
+        break;
+      case typeof arg2 === 'undefined':
+        details = transactionsDetail;
+        break;
+      default:
+        throw Error(`wrong input parameters: ${arg2}, ${transactionsDetail}`);
+    }
```

Comment:
```suggestion
    const details = arg2 === undefined || Array.isArray(arg2) ? transactionsDetail : arg2;
```
This omits the explicit error but parameter validation is rarely used in the library and the code will fail anyway for faulty parameters.